packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -456,52 +458,67 @@ class _BottomNavigationTile extends StatelessWidget {
+            _TileIcon(
+              colorTween: colorTween,
+              animation: animation,
+              iconSize: iconSize,
+              selected: selected,
+              item: item,
+              selectedIconTheme: selectedIconTheme ?? bottomTheme.selectedIconTheme,
+              unselectedIconTheme: unselectedIconTheme ?? bottomTheme.unselectedIconTheme,
+            ),
+            _Label(
+              colorTween: colorTween,
+              animation: animation,
+              item: item,
+              selectedLabelStyle: selectedLabelStyle ?? bottomTheme.selectedLabelStyle,
+              unselectedLabelStyle: unselectedLabelStyle ?? bottomTheme.unselectedLabelStyle,
+              showSelectedLabels: showSelectedLabels ?? bottomTheme.showUnselectedLabels,
+              showUnselectedLabels: showUnselectedLabels ?? bottomTheme.showUnselectedLabels,
+            ),
+          ],
+        ),
+      ),
+    );
+
+    if (item.label != null) {
+      result = Tooltip(
+        message: item.label,
+        preferBelow: false,
+        verticalOffset: selectedIconSize + selectedFontSize,
+        child: result,
+      );
+    }
+
+    result = MergeSemantics(
       child: Semantics(
-        container: true,
         selected: selected,
         child: Stack(
           children: <Widget>[
-            InkResponse(
-              onTap: onTap,
-              mouseCursor: mouseCursor,
-              child: Padding(
-                padding: EdgeInsets.only(top: topPadding, bottom: bottomPadding),
-                child: Column(
-                  crossAxisAlignment: CrossAxisAlignment.center,
-                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
-                  mainAxisSize: MainAxisSize.min,
-                  children: <Widget>[
-                    _TileIcon(
-                      colorTween: colorTween,
-                      animation: animation,
-                      iconSize: iconSize,
-                      selected: selected,
-                      item: item,
-                      selectedIconTheme: selectedIconTheme ?? bottomTheme.selectedIconTheme,
-                      unselectedIconTheme: unselectedIconTheme ?? bottomTheme.unselectedIconTheme,
-                    ),
-                    _Label(
-                      colorTween: colorTween,
-                      animation: animation,
-                      item: item,
-                      selectedLabelStyle: selectedLabelStyle ?? bottomTheme.selectedLabelStyle,
-                      unselectedLabelStyle: unselectedLabelStyle ?? bottomTheme.unselectedLabelStyle,
-                      showSelectedLabels: showSelectedLabels ?? bottomTheme.showUnselectedLabels,
-                      showUnselectedLabels: showUnselectedLabels ?? bottomTheme.showUnselectedLabels,
-                    ),
-                  ],
-                ),
-              ),
-            ),
+            result,
             Semantics(
+              container: true,
```

Comment:
Since restructuring the PR, is this still needed?

Reply:
I simplified the tree based on goderbauer's comment: https://github.com/flutter/flutter/pull/59127/files#r450472105